src/Transaction.jsx

```diff
@@ -1,36 +1,43 @@
+import Decimal from 'decimal.js-light';
 import moment from 'moment';
 
 import LoonyInternalError from './LoonyInternalError';
 
 export default class Transaction {
   constructor({ id = null, account = '', dateMs = 0, payee = '', category = '',
-                memo = '', outflow = 0.0, inflow = 0.0 }) {
+                memo = '', outflow = new Decimal('0'),
+                inflow = new Decimal('0') }) {
     this.id = id;
     this.account = account;
     this.dateMs = dateMs;
     this.payee = payee;
     this.category = category;
     this.memo = memo;
 
-    if (outflow !== 0.0 && inflow !== 0.0) {
+    if (!(inflow instanceof Decimal)) {
+      throw new LoonyInternalError(`inflow must be Decimal: ${inflow}`);
+    }
+    if (!(outflow instanceof Decimal)) {
+      throw new LoonyInternalError(`outflow must be Decimal: ${outflow}`);
+    }
+
+    if (!outflow.isZero() && !inflow.isZero()) {
       throw new LoonyInternalError(
           `Both outflow=${outflow} and inflow=${inflow} can't be specified`);
     }
-    if (outflow < 0) {
+    if (outflow.isNegative()) {
       throw new LoonyInternalError(`outflow can't be negative: ${outflow}`);
     }
-    if (inflow < 0) {
+    if (inflow.isNegative()) {
       throw new LoonyInternalError(`inflow can't be negative: ${outflow}`);
     }
 
-    if (inflow === 0 && outflow === 0) {
-      this.amountMinor = 0;
-    } else if (inflow > 0) {
-      // TODO: Stop using floats for money.
-      this.amountMinor = inflow * 100;
-    } else if (outflow > 0) {
-      // TODO: Stop using floats for money.
-      this.amountMinor = -1 * outflow * 100;
+    if (inflow.isZero() && outflow.isZero()) {
+      this.amountMinor = new Decimal('0');
```

Comment:
Is there a reason to be doing the `int <-> Decimal` type conversion here?  I think this goes better when it gets written to/read from Firebase.